cpp/src/arrow/util/cpu_info.cc

```diff
@@ -59,17 +59,75 @@ void __cpuidex(int CPUInfo[4], int function_id, int subfunction_id) {
 }
 #endif
 
+#if defined(__GNUC__) && defined(__linux__) && defined(__aarch64__)
+// There is no direct instruction to get cache size on Arm64 like '__cpuid' on x86;
+// Get Arm64 cache size by reading '/sys/devices/system/cpu/cpu0/cache/index*/size';
+// index* :
+//   index0: L1 Dcache
+//   index1: L1 Icache
+//   index2: L2 cache
+//   index3: L3 cache
+#include <errno.h>
+bool get_cache_size_arm64(const char* filename, int* size) {
+  bool cache_supported = false;
+  char* content = nullptr;
+  char* last_char = nullptr;
+  size_t file_len = 0;
+  int cardinal_num = 0;
+  int multip = 0;
+
+  if (size == nullptr) return cache_supported;
```

Comment:
Our preferred C++ style is more like:
```c++
if (size == nullptr) {
  return cache_supported;
}
```

(i.e. always use braces `{...}` after `if`)